exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/StreamingAggTemplate.java

```diff
@@ -297,16 +413,33 @@ private final void resetIndex() {
     incIndex();
   }
 
-  private final AggOutcome setOkAndReturn() {
-    if (first) {
-      this.outcome = IterOutcome.OK_NEW_SCHEMA;
+  /**
+   * Set the outcome to OK (or OK_NEW_SCHEMA) and return the AggOutcome parameter
+   * @param outcome
+   * @return outcome
+   */
+  private final AggOutcome setOkAndReturn( IterOutcome outcome) {
```

Comment:
A suggestion: Split off another method - *setOkAndReturnForEmit()* - to be used when (outcome == EMIT). This would yield two simpler methods (fewer condition checks).  As in all places the current method is used, the value of "outcome" is known (EMIT or anything-else), then just call the appropriate one.